infra/discovery/src/listener/listener.js

```diff
@@ -134,84 +136,92 @@ async function main() {
+      // out events outside of interval (processedToBlock, toBlock].
+      const newEvents = [].concat(
+        ...eventArrays.filter(event => {
+          if (
+            event &&
+            event.blockNumber > processedToBlock &&
+            event.blockNumber <= toBlock
+          ) {
+            return event
+          }
+        })
+      )
 
-    // Filter out events outside of interval (processedToBlock, toBlock].
-    let newEvents = events.filter(
-      event =>
-        event.blockNumber > processedToBlock && event.blockNumber <= toBlock
-    )
-    logger.debug(`Got ${newEvents.length} new events`)
-
-    if (context.config.concurrency > 1) {
-      // Concurrency greater than 1 -> Backfill mode.
-      const limit = pLimit(context.config.concurrency)
-      const promises = []
-      newEvents.forEach(newEvent => {
-        promises.push(limit(() => handleEvent(newEvent, context)))
-      })
-      await Promise.all(promises)
-    } else {
-      // Concurrency set to 1 -> Normal operation mode.
-      // Sort events by blockNumber and logIndex to process them in order.
-      // In normal operation (not backfill), ordering does matter for handlers.
-      // For example assume a buyer updates their identity to add their email
-      // then makes a purchase. If the handler would process the purchase
-      // event before the identity update, we may not be able to send the buyer
-      // a confirmation email about their offer.
-      newEvents = sortBy(newEvents, ['blockNumber', 'logIndex'], ['asc', 'asc'])
-
-      // Have the handler process each event.
-      for (const event of newEvents) {
-        // Note: we purposely do not set the exitOnError option of withRetrys to false.
-        // In case all retries fails, it indicates something is wrong at the system
-        // level and a process restart may fix it.
-        await withRetrys(async () => {
-          return handleEvent(event, context)
+      logger.debug(`Got ${newEvents.length} new events for ${contractKey}`)
+
+      if (context.config.concurrency > 1) {
+        // Concurrency greater than 1 -> Backfill mode.
+        const limit = pLimit(context.config.concurrency)
+        const promises = []
+        newEvents.forEach(newEvent => {
+          promises.push(limit(() => handleEvent(newEvent, context)))
+        })
+        await Promise.all(promises).catch(err => {
+          logger.error(`Unhandled error in event handler`)
+          logger.error(err)
         })
+      } else {
+        // Have the handler process each event.
+        for (const event of newEvents) {
+          // Note: we purposely do not set the exitOnError option of withRetrys to false.
+          // In case all retries fails, it indicates something is wrong at the system
+          // level and a process restart may fix it.
+          await withRetrys(async () => {
+            try {
+              return handleEvent(event, context)
+            } catch (err) {
```

Comment:
Catching error here will prevent retries. Let's let withRetrys logging the error (it should already do that I think) ?